### PR TITLE
Update URLs to be https and skip redirects

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Update URLs to be https and skip redirects to improve security and performance.

Superseeds https://github.com/mozilla/addons-server/pull/3312

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
Rebased to master (didn't have 1.8 changes) should make circleci succeed now.

File: src/olympia/templates/copyright.html [hunk 1]
```diff
@@ -22,9 +22,9 @@
   <p>
     {% trans %}
       Except where otherwise
-      <a href="http://www.mozilla.org/about/legal.html#site">noted</a>,
+      <a href="https://www.mozilla.org/about/legal/#site">noted</a>,
```
Comment (muffinresearch) [inline src/olympia/templates/copyright.html]: I don't see a site fragment in the linked page.

Comment (EnTeQuAk) [inline src/olympia/templates/copyright.html]: Good point, fixed in c3a11c1

Comment (tofumatt) [inline src/olympia/templates/copyright.html]: I'm guessing it's been moved to https://www.mozilla.org/en-US/about/legal/terms/mozilla/, which seems specifically about websites.

**Comment by muffinresearch** [comment]
r+ :shipit:

**Pull request merged**